Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1178,13 +1180,32 @@ void RenderBlockFlow::determineLogicalLeftPositionForChild(RenderBox& child, App
         startPosition += (style().isLeftToRightDirection() ? 1 : -1) * verticalScrollbarWidth();
     if (style().scrollbarGutter().bothEdges && !isHorizontalWritingMode())
         startPosition += (style().isLeftToRightDirection() ? 1 : -1) * horizontalScrollbarHeight();
-    LayoutUnit totalAvailableLogicalWidth = borderAndPaddingLogicalWidth() + availableLogicalWidth();
+    const LayoutUnit totalAvailableLogicalWidth = borderAndPaddingLogicalWidth() + availableLogicalWidth();
+
+    auto computeSelfAlignedLogicalLeftPositionForChild = [&](LayoutUnit currentPosition) -> LayoutUnit {
+        auto startAlignedPosition = currentPosition;
+        auto endAlignedPosition = totalAvailableLogicalWidth - startAlignedPosition - logicalWidthForChild(child);
+        const auto selfAlignmentStyle = child.style().resolvedJustifySelf(&style(), ItemPosition::Normal);
+        const auto resolvedDirection = selfAlignmentStyle.isSelfDirectioned() ? child.style().direction() : style().direction();
+        if (!isLeftToRightDirection(resolvedDirection))
+            std::swap(startAlignedPosition, endAlignedPosition);
+        if (child.style().justifySelf().overflow() == OverflowAlignment::Safe && logicalWidthForChild(child) > totalAvailableLogicalWidth)
+            return startAlignedPosition;
+        if (selfAlignmentStyle.isStartward(resolvedDirection) || selfAlignmentStyle.position() == ItemPosition::Stretch)
+            return startAlignedPosition;
+        if (selfAlignmentStyle.isEndward(resolvedDirection))
+            return endAlignedPosition;
+        if (selfAlignmentStyle.isCentered())
+            return totalAvailableLogicalWidth / 2 - logicalWidthForChild(child) / 2;
+        ASSERT_NOT_REACHED();
+        return { };
+    };
 
     LayoutUnit childMarginStart = marginStartForChild(child);
     LayoutUnit newPosition = startPosition + childMarginStart;
 
-    LayoutUnit positionToAvoidFloats;
-    
+    LayoutUnit positionToAvoidFloats { };
```

Comment:
any particular reason for this change?

Reply:
For a reason I can't remember, I thought LayoutUnit was a typedef'd float so I thought I shouldn't leave it uninitialized. Since it's actually a class type I should revert this for consistency.

Reply:
yeah, InlineLayoutUnit is the typedef'd float